File: app/src/main/java/com/example/alockwood/windowinsetsplayground/CollapsingToolbarFragment.java
package com.example.alockwood.windowinsetsplayground;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.CollapsingToolbarLayout;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

public class CollapsingToolbarFragment extends BaseFragment {

  @Override
  public View onCreateView(
      @NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    final View view = inflater.inflate(R.layout.fragment_collapsing_toolbar, container, false);

    final CollapsingToolbarLayout collapsingToolbar = view.findViewById(R.id.collapsing_toolbar);
    collapsingToolbar.setTitle(getString(R.string.collapsing_toolbar));

    final ImageView profilePhoto = view.findViewById(R.id.backdrop_photo);
    Glide.with(this).load(Constants.COVER_PHOTO_URL).into(profilePhoto);

    return view;
  }

  @Override
  protected int getStatusBarBackground() {
    return android.R.color.transparent;
  }

  @Override
  protected boolean shouldUseLightStatusBar() {
    return false;
  }

  @Override
  protected boolean shouldDrawContentUnderStatusBar() {
    return true;
  }
}

File: app/src/main/java/com/example/alockwood/windowinsetsplayground/CollapsingToolbarFragment.kt
package com.example.alockwood.windowinsetsplayground

import android.os.Bundle
import android.support.design.widget.CollapsingToolbarLayout
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView

import com.bumptech.glide.Glide

class CollapsingToolbarFragment : Fragment() {

    override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_collapsing_toolbar, container, false)

        val collapsingToolbar = view.findViewById<CollapsingToolbarLayout>(R.id.collapsing_toolbar)
        collapsingToolbar.title = getString(R.string.collapsing_toolbar)

        val profilePhoto = view.findViewById<ImageView>(R.id.backdrop_photo)
        Glide.with(this).load(Constants.COVER_PHOTO_URL).into(profilePhoto)

        return view
    }
}
